pkg/downloader/chart_downloader.go

```diff
@@ -262,6 +262,7 @@ func (c *ChartDownloader) ResolveChartVersion(ref, version string) (*url.URL, er
 		}
 		q := repoURL.Query()
 		// We need a trailing slash for ResolveReference to work, but make sure there isn't already one
+		repoURL.RawPath = strings.TrimSuffix(repoURL.RawPath, "/") + "/"
```

Comment:
https://github.com/helm/helm/pull/9763#discussion_r644001942

Reply:
This doesn't work (when the URL has a query string):

```
    chart_downloader_test.go:79: reference, querystring repo: expected http://example.com/alpine-1.2.3.tgz?key=value, got http://example.com/alpine-1.2.3.tgz?key=value%2F
```

```diff
diff --git a/pkg/downloader/chart_downloader.go b/pkg/downloader/chart_downloader.go
index 6134c0e6..bb1ba90a 100644
--- a/pkg/downloader/chart_downloader.go
+++ b/pkg/downloader/chart_downloader.go
@@ -256,14 +256,12 @@ func (c *ChartDownloader) ResolveChartVersion(ref, version string) (*url.URL, er
 
        // If the URL is relative (no scheme), prepend the chart repo's base URL
        if !u.IsAbs() {
-               repoURL, err := url.Parse(rc.URL)
+               // We need a trailing slash for ResolveReference to work, but make sure there isn't already one
+               repoURL, err := url.Parse(strings.TrimSuffix(rc.URL, "/") + "/")
                if err != nil {
                        return repoURL, err
                }
                q := repoURL.Query()
-               // We need a trailing slash for ResolveReference to work, but make sure there isn't already one
-               repoURL.RawPath = strings.TrimSuffix(repoURL.RawPath, "/") + "/"
-               repoURL.Path = strings.TrimSuffix(repoURL.Path, "/") + "/"
                u = repoURL.ResolveReference(u)
                u.RawQuery = q.Encode()
                // TODO add user-agent
```

Reply:
@cndoit18 Please review again...